AudioApplication.cpp

```diff
@@ -0,0 +1,46 @@
+#include "AudioApplication.h"
+#include <iostream>
+
+AudioApplication::AudioApplication(float setDefaultRatio)
+{
+    initRatio = setDefaultRatio;
+}
+
+AudioApplication::~AudioApplication()
+{
+    std::cout<< "Closing Audio Application" << std::endl;
+}
+
+float AudioApplication::compress(float input, float ratio, float threshold)
+{
+    if (input>threshold)
+    {
+        std::cout << "Signal Compressed" << std::endl;
+        return input * (0.1f*ratio);
+    } 
+    else
+    {
+        return input;
+    }
+}
```

Comment:
If you find yourself writing code like this:
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    else 
    {
        return anotherValue;
    }
}
```
if the `if()` block is entered, the `return aValue;` is the last line of code to be executed.
you have no code after the `else` block that needs executing, since your `else` block contains a `return` statement.
`return` statements are the last thing that **ever happens** in a function
That means if your `if()` block is **NOT** entered, it will be skipped.  What is the next thing that happens?  `return anotherValue;`
that means you do **NOT** need the else block, and it is just causing visual clutter.
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    return anotherValue;
}
```
- [ ] revise